src/components/CreateOrModifyTagDialog.tsx

```diff
@@ -1,132 +1,145 @@
-                    <div className="flex justify-between items-center">
-                      <FormLabel>Color</FormLabel>
-                      <div className="flex items-center gap-2">
-                        <p className="font-mono">{field.value}</p>
-                        <FormControl>
-                          <Input
-                            type="color"
-                            {...field}
-                            className="w-[50px] cursor-pointer"
-                          />
-                        </FormControl>
-                      </div>
-                    </div>
-                    <FormMessage />
-                  </FormItem>
-                )}
-              />
-              <DialogFooter>
-                <Button type="submit">Save tag</Button>
-              </DialogFooter>
-            </form>
-          </Form>
-        </div>
-      </DialogContent>
-    </Dialog>
-  );
+	function onSubmit(formValues: CreateOrModifyTagFormSchema) {
+		currentTagId === null
+			? createTagHandler(formValues)
+			: updateTagHandler(formValues);
+	}
+
+	return (
+		<Dialog
+			open={isModalOpen}
+			onOpenChange={() => {
+				closeModal();
+			}}
+		>
+			<DialogContent className="sm:max-w-[425px]">
+				<DialogHeader>
+					<DialogTitle>
+						{(modalState === 'ADD_TAG' && 'Create new tag') ||
+							(modalState === 'EDIT_TAG' && 'Edit tag')}
+					</DialogTitle>
+				</DialogHeader>
+				<div className="grid gap-4 py-4">
+					<Form {...form}>
+						<form onSubmit={form.handleSubmit(onSubmit)} className="space-y-8">
+							<FormField
+								control={form.control}
+								name="name"
+								render={({ field }) => (
+									<FormItem>
+										<FormLabel>Name</FormLabel>
+										<FormControl>
+											<Input placeholder="" {...field} />
+										</FormControl>
+										<FormMessage />
+									</FormItem>
+								)}
+							/>
+							<FormField
+								control={form.control}
+								name="description"
+								render={({ field }) => (
+									<FormItem>
+										<FormLabel>Description</FormLabel>
+										<FormControl>
+											<Textarea placeholder="" {...field} />
+										</FormControl>
+										<FormMessage />
+									</FormItem>
+								)}
+							/>
+							<FormField
+								control={form.control}
+								name="color"
+								render={({ field }) => (
+									<FormItem>
+										<div className="flex items-center justify-between">
+											<FormLabel>Color</FormLabel>
+											<div className="flex items-center gap-2">
+												<p className="font-mono">{field.value}</p>
+												<FormControl>
+													<Input
+														type="color"
+														{...field}
+														className="w-[50px] cursor-pointer"
+													/>
+												</FormControl>
+											</div>
+										</div>
+										<FormMessage />
+									</FormItem>
+								)}
+							/>
+							<DialogFooter>
+								<Button type="submit">Save tag</Button>
```

Comment:
You can add dictionary to the project

Reply:
The dictionary also will be added in another task